plenum/server/replica.py

```diff
@@ -1480,31 +1495,44 @@ def checkIfCheckpointStable(self, key: Tuple[int, int]):
             return False
 
     def stashCheckpoint(self, ck: Checkpoint, sender: str):
+        logger.debug('{} stashing {} from {}'.format(self, ck, sender))
         seqNoStart, seqNoEnd = ck.seqNoStart, ck.seqNoEnd
-        if (seqNoStart, seqNoEnd) not in self.stashedRecvdCheckpoints:
-            self.stashedRecvdCheckpoints[seqNoStart, seqNoEnd] = {}
-        self.stashedRecvdCheckpoints[seqNoStart, seqNoEnd][sender] = ck
+        if ck.viewNo not in self.stashedRecvdCheckpoints:
+            self.stashedRecvdCheckpoints[ck.viewNo] = {}
+        stashed_for_view = self.stashedRecvdCheckpoints[ck.viewNo]
+        if (seqNoStart, seqNoEnd) not in stashed_for_view:
+            stashed_for_view[seqNoStart, seqNoEnd] = {}
+            stashed_for_view[seqNoStart, seqNoEnd][sender] = ck
```

Comment:
Is there an extra indentation for `stashed_for_view[seqNoStart, seqNoEnd][sender] = ck`?